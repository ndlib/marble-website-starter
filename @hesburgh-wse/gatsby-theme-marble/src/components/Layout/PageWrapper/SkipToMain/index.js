import React from 'react'
import style from './style.module.css'

const SkipToMain = () => {
  return (
    <a className={style.skipToMain} href='#mainContent'>Skip to main content.</a>
  )
}

export default SkipToMain
