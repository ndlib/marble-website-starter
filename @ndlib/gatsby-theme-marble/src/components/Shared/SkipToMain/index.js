/** @jsx jsx */
// eslint-disable-next-line no-unused-vars
import React from 'react'
import PropTypes from 'prop-types'
import { jsx } from 'theme-ui'
import Link from 'components/Shared/Link'

const active = {
  color: '#fff',
  backgroundColor: '#000',
  left: 'auto',
  top: 'auto',
  width: '30%',
  height: 'auto',
  overflow: 'auto',
  margin: '10px 35%',
  padding: '5px',
  textAlign: 'center',
  fontSize: '1.2em',
  zIndex: '999',
}

export const sx = {
  left: '-999px',
  position: 'absolute',
  top: 'auto',
  width: '1px',
  height: '1px',
  overflow: 'hidden',
  zIndex: '-999',
  '&:focus': active,
  '&:active': active,
}

const SkipToMain = ({ location }) => {
  return (
    <Link sx={sx} id='skipToMain' to={location.pathname + '?scrollto=mainContent'}>Skip to main content.</Link>
  )
}

SkipToMain.propTypes = {
  location: PropTypes.object,
}

export default SkipToMain
