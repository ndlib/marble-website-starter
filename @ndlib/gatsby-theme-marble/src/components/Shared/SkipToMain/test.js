import React from 'react'
import renderer from 'react-test-renderer'

import SkipToMain from './'

test('SkipToMain', () => {
  const tree = renderer
    .create(<SkipToMain location={{ pathname: 'path' }} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
