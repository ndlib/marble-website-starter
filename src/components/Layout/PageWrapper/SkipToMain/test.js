import React from 'react'
import { shallow } from 'enzyme'
import SkipToMain from './'

test('SkipToMain', () => {
  const wrapper = shallow(<SkipToMain />)
  expect(wrapper.find('a.skipToMain').props().href).toEqual('#mainContent')
  expect(wrapper.find('a.skipToMain').text()).toEqual('Skip to main content.')
})
