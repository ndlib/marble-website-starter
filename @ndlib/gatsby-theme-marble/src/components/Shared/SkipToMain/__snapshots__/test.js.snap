// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipToMain 1`] = `
<a
  className="css-gb7gys-SkipToMain"
  href="#mainContent"
  id="skipToMain"
>
  Skip to main content.
</a>
`;
